explorer/client/src/reducers/search.ts

```diff
@@ -4,30 +4,19 @@ export interface IState {
 
 export type Query = string | undefined
 
-export type SearchAction =
-  | { type: 'UPDATE_SEARCH_QUERY'; query?: string }
-  | { type: '@@redux/INIT' }
-  | { type: '@@INIT' }
+export type Action = { type: string; location?: Location }
```

Comment:
This feels a little gross as it matches all actions. I couldn't think of a better way to inject `location` for testing. Open to suggestions.

Reply:
I'm just happy we've reached a point where we can all agree `string` type that matches all actions is gross.